### ISSUE TITLE
TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by Claud** [nickjj/ansible-docker#41]
It is my host_vars:

```
docker__registries:
  - registry_url: "..."
    username: "...."
    password: "....."
```

I see this error:

```
TASK [nickjj.docker : Manage login credentials for 1 or more Docker registries] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: 'NoneType' object is not iterable
fatal: [192.168.142.5]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
```

Ansible version:

```
ansible 2.7.1
  config file = None
  configured module search path = [u'/home/claud/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:32:47) [GCC 4.8.4]

```

**Comment by nickjj** [comment]
Hi,

What version of this role are you using?

**Comment by Claud** [comment]
> {install_date: 'Thu Nov  8 21:03:28 2018', version: v1.4.0}

**Comment by nickjj** [comment]
Have you tried logging into the registry manually using `docker login` and using the exact same credentials you supplied to the role, using the same exact user?

**Comment by Claud** [comment]
yes. I tried this.

```

WARNING! Your password will be stored unencrypted in /home/claud/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded

```

**Comment by nickjj** [comment]
What happens if you set `system_user: "root"` to your registry list item?

Or replace root with a user that exists in `docker__users`.

**Comment by Claud** [comment]
I have set `system_user: "root"` and this worked.

**Comment by Claud** [comment]
Thx

**Comment by nickjj** [comment]
Thanks for the report. This was actually a bug then.

I need to have the system_user default to root if no system_user is supplied.

**Comment by nickjj** [comment]
This has been fixed in v1.5.0 and pushed to the Galaxy.

The new behavior is:

1. Supplied your own `system_user`? It will use that.
2. OR Have at least 1 user set in `docker__users`? It will use the first user in that list.
3. OR Didn't have a user set? Fall back to using the root user.

In your case it means you can remove `system_user: "root"` since it will use that by default.

**Comment by Claud** [comment]
No, I can't do it. I have "docker__users" and it has access to "sudo" command. But when I tried 
 to set it for "system_user" I see an error.
